scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,151 @@
+
+This generator uses IPC-7351B equations and fillet definitions to derive a footprint for gullwing style packages. (fillet definitions can be changed by pointing to a different definition file.) The script generates footprints in zero orientation version A (pin 1 at top left corner).
+Pads are generated using rounded rectangle pads as suggested in preliminary releases of IPC-7351C.
+
+Examples of supported packages include: QFN, SOIC, SO.
+
+## Running the script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_gullwing_generator.py size_definitions/eqfp.yaml` (replace `eqfp.yaml` with the size definition file that contains your part.)
+
+### Optional script parameters
+
+* --global_config: the config file defining how the footprint will look like. (KLC)' (default='../../tools/global_config_files/config_KLCv3.0.yaml')
+* --series_config: the config file defining series parameters (footprint naming). (default='../package_config_KLCv3.yaml')
+
+* --density: Density level (L,N,M) (default='N')
+* --ipc_doc: IPC definition document (default='../ipc_definitions.yaml')
+
+* --force_rectangle_pads: Force the generation of rectangle pads instead of rounded rectangle
+* --kicad4_compatible: Create footprints compatible with version 4 (avoids round-rect and custom pads).
+
+## Size definition format
+
+Every file contains a header with parameters applied to all parts. These define the common footprint name prefix and the output library.
+
+``` yaml
+FileHeader:
+  library_Suffix: 'QFP' #resulting library name Package_QFP
+  device_type: 'EQFP' #footprint names will be of style ...EQFP-pincount_...
+```
+
+---
+
+Every further entry in the script is assumed to be a package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was generated by which parameter set.)
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and naming parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- Footprint name generation control
+  - Manufacturer name. Will be added as a prefix if given. (`manufacturer`) {string}
+  - Part number. Will be added as a prefix if given. (`part_number`) {string}
+  - Suffix: A custom suffix (added after pin pitch in default naming format). Can include parameters `pad_x` and `pad_y`.
+  - Custom naming (`custom_name_format`) {python format string}
+    - The full default format string is '{man:s}_{mpn:s}_{pkg:s}-{pincount:d}-1EP_{size_x:g}x{size_y:g}mm_P{pitch:g}mm{suffix:s}_EP{ep_size_x:g}x{ep_size_y:g}mm_Mask{mask_size_x:g}x{mask_size_y:g}mm{suffix2:s}{vias:s}' (The same parameters can be used in your custom format. Exposed pad parameters are not available for components without exposed pad.)
```

Comment:
Should this string be enclosed in code marker since the underscores make the mpn section appear in italiac and mess up the intended rendering of the format?

Reply:
done